proto-public/pbmesh/v1alpha1/pbproxystate/traffic_permissions.proto

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v1alpha1.pbproxystate;
+
+message L7TrafficPermission {}
+
+enum TrafficPermissionAction {
+  INTENTION_ACTION_DENY = 0;
+  INTENTION_ACTION_ALLOW = 1;
+}
+
+message L4TrafficPermission {
+  TrafficPermissionAction default_action = 1;
+  // permission is constructed by pre-flattening all relevant permissions.
+  L4Permission permission = 2;
+}
+
+message L4Permission {
+  repeated L4Source sources = 1;
+  repeated L4Source exclude_sources = 2;
```

Comment:
I think this structure would not translate directly from our API. We want to be able to express an `or` of sources with each source potentially `and`'ed with a not of exclude sources. 

This proposed structure would allow us to do `(source1 or source2) and (not source3 or not source4)`.

What if I want to say:

`source1 or (source2 and not source3)`?

Reply:
I'd make it something like this:

```
message L4Permission {
  repeated Source sources = 1;
}

message Source {
  string principal = 1;
  repeated string exclude_principals = 2;
}
```

Reply:
`source1 or (source2 and not source3)` would be:
```json
[
  {
    "sources": "source1"
  },
  {
    "sources": "source2",
    "exclude_sources": ["source3"]
  }
]
```

For TCP services, only need to match a single spiffe, so I changed the type to be a `string`. Good call 👍